cuda/matrix/ell_kernels.cu

```diff
@@ -336,10 +337,69 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_ELL_ADVANCED_SPMV_KERNEL);
 
 
+namespace kernel {
+
+
+template <typename ValueType>
+__global__ __launch_bounds__(cuda_config::warp_size) void initialize_zero_dense(
+    size_type num_rows, size_type num_cols, size_type stride,
+    ValueType *__restrict__ result)
+{
+    const auto tidx_x = threadIdx.x + blockDim.x * blockIdx.x;
+    const auto tidx_y = threadIdx.y + blockDim.y * blockIdx.y;
+    if (tidx_x < num_cols && tidx_y < num_rows) {
+        result[tidx_y * stride + tidx_x] = zero<ValueType>();
+    }
+}
+
+
+template <typename ValueType, typename IndexType>
+__global__ __launch_bounds__(default_block_size) void fill_in_dense(
+    size_type num_rows, size_type nnz, size_type source_stride,
+    const IndexType *__restrict__ col_idxs,
+    const ValueType *__restrict__ values, size_type result_stride,
+    ValueType *__restrict__ result)
+{
+    const auto tidx = threadIdx.x + blockIdx.x * blockDim.x;
+    if (tidx < num_rows) {
+        for (auto col = 0; col < nnz; col++) {
+            result[tidx * result_stride +
+                   col_idxs[tidx + col * source_stride]] +=
+                values[tidx + col * source_stride];
```

Comment:
Why `+=`? Is it possible that there is already a non-zero value in `result`?

Reply:
Just to answer: it is in case you access padded values, which are always zero. `+=` here is to prevent using an `if` to check if the value in `values` is zero.